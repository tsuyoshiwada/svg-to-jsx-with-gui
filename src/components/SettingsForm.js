import React from 'react';
import styled from 'styled-components';
import { Flex, Box } from 'grid-styled';
import { dripForm } from 'react-drip-form';
import { FieldGroup, Input, Checkbox } from 'react-drip-form-components';

const Field = styled.div`
  & label {
    display: block;
    margin: 0 0 0.2em;
    font-weight: bold;
    cursor: pointer;
  }
`;

const singleCheckbox = v => !!v;

export const SettingsForm = () => (
  <div>
    <h2>Settings</h2>

    <h3>SVG <small>svg transformer</small></h3>
    <FieldGroup name="svgoPlugins" multiple>
      <Checkbox value="cleanupIDs">Clean IDs</Checkbox>
      <Checkbox value="cleanupListOfValues">Round / rewrite number lists</Checkbox>
      <Checkbox value="cleanupNumericValues">Round / rewrite numbers</Checkbox>
      <Checkbox value="convertColors">Minify colours</Checkbox>
      <Checkbox value="collapseGroups">Collapse useless groups</Checkbox>
      <Checkbox value="convertPathData">Round/rewrite paths</Checkbox>
      <Checkbox value="convertShapeToPath">Shapes to (smaller) paths</Checkbox>
      <Checkbox value="mergePaths">Merge paths</Checkbox>
      <Checkbox value="removeDimensions">Prefer viewBox to width / height</Checkbox>
      <Checkbox value="removeDesc">Remove <code>&lt;desc&gt;</code></Checkbox>
      <Checkbox value="removeDoctype">Remove doctype</Checkbox>
      <Checkbox value="removeEditorsNSData">Remove editor data</Checkbox>
      <Checkbox value="removeEmptyAttrs">Remove empty attrs</Checkbox>
      <Checkbox value="removeEmptyContainers">Remove empty containers</Checkbox>
      <Checkbox value="removeEmptyText">Remove empty text</Checkbox>
      <Checkbox value="removeHiddenElems">Remove hidden elements</Checkbox>
      <Checkbox value="removeMetadata">Remove <code>&lt;metadata&gt;</code></Checkbox>
      <Checkbox value="removeTitle">Remove <code>&lt;title&gt;</code></Checkbox>
      <Checkbox value="removeUselessDefs">Remove unused defs</Checkbox>
      <Checkbox value="removeUselessStrokeAndFill">Remove useless stroke & fill</Checkbox>
      <Checkbox value="removeViewBox">Remove viewBox</Checkbox>
      <Checkbox value="removeXMLProcInst">Remove XML instructions</Checkbox>
    </FieldGroup>

    <h3>Editor <small>powered by <a href="https://github.com/securingsincity/react-ace">react-ace</a></small></h3>
    <Flex mx={-2} mb={2}>
      <Box width={1/2} px={2}>
        <Field>
          <label htmlFor="editor-fontSize">Font size</label>
          <Input
            type="number"
            name="editor.fontSize"
            id="editor-fontSize"
            label="Font size"
          />
        </Field>
      </Box>
      <Box width={1/2} px={2}>
        <Field>
          <label htmlFor="editor-tabSize">Tab size</label>
          <Input
            type="number"
            id="editor-tabSize"
            name="editor.tabSize"
            label="Tab size"
          />
        </Field>
      </Box>
    </Flex>
    <Checkbox name="editor.useSoftTabs" value={true} parser={singleCheckbox}>Soft tab</Checkbox>
    <Checkbox name="editor.showGutter" value={true} parser={singleCheckbox}>Show Line Numbers</Checkbox>
  </div>
);

export default dripForm({
  validations: {
    'editor.fontSize': { required: true, number: true },
    'editor.tabSize': { required: true, number: true },
  },
  normalizers: {
    'editor.fontSize': { toInt: true, between: { min: 12, max: 30 } },
    'editor.tabSize': { toInt: true, between: { min: 2, max: 8 } },
  },
})(SettingsForm);
