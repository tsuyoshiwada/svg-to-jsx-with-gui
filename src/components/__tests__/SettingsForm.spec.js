import React from 'react';
import { shallow } from 'enzyme';
import { SettingsForm } from '../SettingsForm';


describe('components#<SettingsForm />', () => {
  test('Should be render', () => {
    const wrapper = shallow(<SettingsForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
