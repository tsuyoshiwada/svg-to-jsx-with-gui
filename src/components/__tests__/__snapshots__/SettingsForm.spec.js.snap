// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components#<SettingsForm /> Should be render 1`] = `
<div>
  <h2>
    Settings
  </h2>
  <h3>
    SVG 
    <small>
      svg transformer
    </small>
  </h3>
  <dripFormGroup(FieldGroup)
    label={null}
    messages={null}
    multiple={true}
    name="svgoPlugins"
    normalizers={null}
    shouldDisplayError={[Function]}
    validations={null}
    value={null}
  >
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="cleanupIDs"
    >
      Clean IDs
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="cleanupListOfValues"
    >
      Round / rewrite number lists
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="cleanupNumericValues"
    >
      Round / rewrite numbers
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="convertColors"
    >
      Minify colours
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="collapseGroups"
    >
      Collapse useless groups
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="convertPathData"
    >
      Round/rewrite paths
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="convertShapeToPath"
    >
      Shapes to (smaller) paths
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="mergePaths"
    >
      Merge paths
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeDesc"
    >
      Remove 
      <code>
        &lt;desc&gt;
      </code>
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeDoctype"
    >
      Remove doctype
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeEditorsNSData"
    >
      Remove editor data
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeEmptyAttrs"
    >
      Remove empty attrs
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeEmptyContainers"
    >
      Remove empty containers
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeEmptyText"
    >
      Remove empty text
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeHiddenElems"
    >
      Remove hidden elements
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeMetadata"
    >
      Remove 
      <code>
        &lt;metadata&gt;
      </code>
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeTitle"
    >
      Remove 
      <code>
        &lt;title&gt;
      </code>
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeUselessDefs"
    >
      Remove unused defs
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeUselessStrokeAndFill"
    >
      Remove useless stroke & fill
    </dripFormField(Checkbox)>
    <dripFormField(Checkbox)
      formatter={[Function]}
      label={null}
      messages={null}
      normalizers={null}
      onBlur={null}
      onChange={null}
      onFocus={null}
      parser={null}
      shouldDisplayError={[Function]}
      shouldDisplaySpinner={[Function]}
      validations={null}
      value="removeXMLProcInst"
    >
      Remove XML instructions
    </dripFormField(Checkbox)>
  </dripFormGroup(FieldGroup)>
  <h3>
    Editor 
    <small>
      powered by 
      <a
        href="https://github.com/securingsincity/react-ace"
      >
        react-ace
      </a>
    </small>
  </h3>
  <Flex
    mb={2}
    mx={-2}
  >
    <Box
      px={2}
      width={0.5}
    >
      <styled.div>
        <label
          htmlFor="editor-fontSize"
        >
          Font size
        </label>
        <dripFormField(Input)
          formatter={[Function]}
          id="editor-fontSize"
          label="Font size"
          messages={null}
          name="editor.fontSize"
          normalizers={null}
          onBlur={null}
          onChange={null}
          onFocus={null}
          parser={null}
          shouldDisplayError={[Function]}
          shouldDisplaySpinner={[Function]}
          type="number"
          validations={null}
        />
      </styled.div>
    </Box>
    <Box
      px={2}
      width={0.5}
    >
      <styled.div>
        <label
          htmlFor="editor-tabSize"
        >
          Tab size
        </label>
        <dripFormField(Input)
          formatter={[Function]}
          id="editor-tabSize"
          label="Tab size"
          messages={null}
          name="editor.tabSize"
          normalizers={null}
          onBlur={null}
          onChange={null}
          onFocus={null}
          parser={null}
          shouldDisplayError={[Function]}
          shouldDisplaySpinner={[Function]}
          type="number"
          validations={null}
        />
      </styled.div>
    </Box>
  </Flex>
  <dripFormField(Checkbox)
    formatter={[Function]}
    label={null}
    messages={null}
    name="editor.useSoftTabs"
    normalizers={null}
    onBlur={null}
    onChange={null}
    onFocus={null}
    parser={[Function]}
    shouldDisplayError={[Function]}
    shouldDisplaySpinner={[Function]}
    validations={null}
    value={true}
  >
    Soft tab
  </dripFormField(Checkbox)>
  <dripFormField(Checkbox)
    formatter={[Function]}
    label={null}
    messages={null}
    name="editor.showGutter"
    normalizers={null}
    onBlur={null}
    onChange={null}
    onFocus={null}
    parser={[Function]}
    shouldDisplayError={[Function]}
    shouldDisplaySpinner={[Function]}
    validations={null}
    value={true}
  >
    Show Line Numbers
  </dripFormField(Checkbox)>
</div>
`;
